inc/namespace.php

```diff
@@ -328,15 +343,25 @@ function get_rekognition_client() : RekognitionClient {
 		'region'  => 'us-east-1',
 	];
 
+	// Compat with S3 Uploads.
+	if ( defined( 'S3_UPLOADS_REGION' ) ) {
+		$client_args['region'] = S3_UPLOADS_REGION;
+	}
 	if ( defined( 'AWS_REKOGNITION_REGION' ) ) {
 		$client_args['region'] = AWS_REKOGNITION_REGION;
 	}
 
-	if ( defined( 'AWS_REKOGNITION_KEY' ) && defined( 'AWS_REKOGNITION_SECRET' ) && defined( 'AWS_REKOGNITION_REGION' ) ) {
+	// Ensure region is supported.
+	if ( ! in_array( $client_args['region'], get_supported_regions(), true ) ) {
+		$client_args['region'] = 'us-east-1';
+	}
+
+	// Use credentials if provided.
+	if ( defined( 'AWS_REKOGNITION_KEY' ) && defined( 'AWS_REKOGNITION_SECRET' ) ) {
 		$client_args['credentials'] = [
 			'key'    => AWS_REKOGNITION_KEY,
 			'secret' => AWS_REKOGNITION_SECRET,
-			'region' => AWS_REKOGNITION_REGION,
+			'region' => $client_args['region'],
```

Comment:
Will this get set if the instance profile is used instead of the constants?

Reply:
Sorry I see what you mean. The instance profile as far as I can tell has access to Rekognition on any region, at least on our stack so presumably it'd use the client's region when authenticating requests but need to confirm that.

Reply:
@roborourke 	correct, when using instance profiles, the client will use whatever region the SDK client is instantiated with. The SDK will then attempt to authenticate using Instance Profiles, IAM Task Roles, etc when an explicit key/secret pair is not provided to the client.

Reply:
Also, fwiw, the `Client::factory` pattern was deprecated in v3 of their SDK: https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/getting-started_migration.html#client-instantiation-uses-the-constructor